code/modules/projectiles/gun.dm

```diff
@@ -257,6 +260,10 @@
 			addtimer(CALLBACK(src, .proc/process_burst, user, target, message, params, zone_override, sprd, randomized_gun_spread, randomized_bonus_spread, rand_spr, i), fire_delay * (i - 1))
 	else
 		if(chambered)
+			if(user.has_trait(TRAIT_PACIFISM))
```

Comment:
shooting something should not check if you have a trait, having a certain trait should pick up when you shoot something.

Then again this is guncode so...